src/com/parse4cn1/command/ParseUploadCommand.java

```diff
@@ -54,6 +54,7 @@ void setUpRequest(ConnectionRequest request) throws ParseException {
 
         if (contentType != null) {
             request.addRequestHeader(ParseConstants.HEADER_CONTENT_TYPE, contentType);
+            addHeader(ParseConstants.HEADER_CONTENT_TYPE, contentType);
```

Comment:
Clear. Thanks for the explanation. I'll first check if adding CONTENT_TYPE as a default header can now be removed (became necessary after the migration to Parse server but I won't be surprised it has been fixed now). If not, Ill merge your proposed change.